http-client/src/test/java/io/airlift/http/client/BasicAuthenticationFilterTest.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.airlift.http.client;
+
+import org.testng.annotations.Test;
+
+import java.net.URI;
+import java.util.Base64;
+
+import static com.google.common.base.Charsets.UTF_8;
+import static io.airlift.http.client.HttpUriBuilder.uriBuilderFrom;
+import static io.airlift.http.client.Request.Builder.prepareGet;
+import static org.eclipse.jetty.http.HttpHeader.AUTHORIZATION;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.assertNull;
+import static org.testng.Assert.assertTrue;
+
+public class BasicAuthenticationFilterTest
+{
+    @Test
+    public void testBasicAuthentication()
+            throws Exception
+    {
+        BasicAuthenticationFilter filter = new BasicAuthenticationFilter((r) -> r.getUri().getPath().startsWith("/private"), "user", "password");
```

Comment:
This line is a bit long. It might be more readable to wrap or introduce a variable for the predicate.